File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/endpoints/message_actions/AddMessageAction.java
package com.pubnub.api.endpoints.message_actions;

import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.message_actions.PNAddMessageActionResult;
import com.pubnub.api.models.consumer.message_actions.PNSavedMessageAction;

public interface AddMessageAction extends Endpoint<PNAddMessageActionResult> {
    AddMessageAction channel(String channel);

    AddMessageAction messageAction(PNSavedMessageAction messageAction);
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/api/endpoints/message_actions/AddMessageAction.kt
package com.pubnub.api.endpoints.message_actions

import com.pubnub.api.Endpoint
import com.pubnub.internal.PubNubImpl
import com.pubnub.api.models.consumer.message_actions.PNAddMessageActionResult
import com.pubnub.internal.endpoints.message_actions.IAddMessageAction

/**
 * @see [PubNubImpl.addMessageAction]
 */
class AddMessageAction internal constructor(addMessageAction: IAddMessageAction) :
    Endpoint<PNAddMessageActionResult>(),
    IAddMessageAction by addMessageAction
